### PR TITLE
docs: generate module docs from spec

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request generates module documentation from the `x/{module}/spec` folder (same as Cosmos SDK). This pull request also sets the foundation for the module spec documentation for the data module and ecocredit module.

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] followed the [documentation writing guidelines](https://github.com/cosmos/cosmos-sdk/blob/master/docs/DOC_WRITING_GUIDELINES.md)
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct `docs:` prefix in the PR title
- [ ] confirmed all author checklist items have been addressed 
- [ ] confirmed that this PR only changes documentation
- [ ] reviewed content for consistency
- [ ] reviewed content for thoroughness
- [ ] reviewed content for spelling and grammar
- [ ] tested instructions (if applicable)

### CODE REVIEW

File: docs/.vuepress/config.js [hunk 1]
```diff
@@ -132,5 +167,10 @@ module.exports = {
   plugins: [
     '@vuepress/plugin-back-to-top',
     '@vuepress/plugin-medium-zoom',
-  ]
+  ],
+  markdown: {
+    extendMarkdown: md => {
+      md.use(require('./markdown-it-gh'))
+    }
+  }
```
Comment (ryanchristo) [inline docs/.vuepress/config.js:175]: This allows code snippets using the same format with cosmos sdk. Slightly modified from the cosmos vuepress theme: https://github.com/cosmos/vuepress-theme-cosmos/blob/develop/markdown-it-gh.js

**Review by clevinson — APPROVED** [review]
ACK. thx @ryanchristo !

File: x/data/spec/05_client.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Client
+
+The `tx` (transaction) and `q` (query) subcommands below give a good illustration
+of how this module can be used in practice:
+
+### Data Transactions
+
+```sh 
+$ regen tx data --help
```
Comment (ruhatch) [inline x/data/spec/05_client.md:9]: Is there a way to get these for free somehow from the actual CLI?

**Review by ruhatch — APPROVED** [review]
LGTM! I wonder if we can auto generate client docs to avoid syncing issues, but that could be tracked in another issue if we want to try that.

**Pull request merged**